contracts/pcv/utils/NamedStaticPCVDepositWrapper.sol

```diff
@@ -0,0 +1,202 @@
+
+        uint256 lastIndex = pcvDeposits.length - 1;
+
+        if (lastIndex != index) {
+            DepositInfo storage lastvalue = pcvDeposits[lastIndex];
+
+            // Move the last value to the index where the value to delete is
+            pcvDeposits[index] = lastvalue;
+        }
+        // Delete the slot where the moved value was stored
+        pcvDeposits.pop();
+
+        balance -= depositBalance;
+        feiReportBalance -= feiDepositBalance;
+
+        emit BalanceUpdate(oldBalance, balance, oldFeiReportBalance, feiReportBalance);
+        emit DepositRemoved(index);
+    }
+
+    /// @notice helper method to add a PCV deposit
+    function _addDeposit(DepositInfo memory newPCVDeposit) internal {
+        require(newPCVDeposit.feiAmount > 0 || newPCVDeposit.usdAmount > 0, "NamedStaticPCVDepositWrapper: must supply either fei or usd amount");
+
+        uint256 oldBalance = balance;
+        uint256 oldFEIBalance = feiReportBalance;
+
+        balance += newPCVDeposit.usdAmount;
+        feiReportBalance += newPCVDeposit.feiAmount;
+        pcvDeposits.push(newPCVDeposit);
+
+        emit DepositAdded(pcvDeposits.length - 1, newPCVDeposit.depositName);
+        emit BalanceUpdate(oldBalance, balance, oldFEIBalance, feiReportBalance);
+    }
+
+    /// @notice helper method to edit a PCV deposit
+    function _editDeposit(
+        uint256 index,
+        string calldata depositName,
+        uint256 usdAmount,
+        uint256 feiAmount,
+        uint256 underlyingTokenAmount,
+        address underlyingToken
+    ) internal {
+        DepositInfo storage updatePCVDeposit = pcvDeposits[index];
+
+        uint256 oldBalance = balance;
+        uint256 oldFEIBalance = feiReportBalance;
+
+        balance = balance - updatePCVDeposit.usdAmount + usdAmount;
+        feiReportBalance = feiReportBalance - updatePCVDeposit.feiAmount + feiAmount;
+
+        updatePCVDeposit.usdAmount = usdAmount + updatePCVDeposit.usdAmount;
+        updatePCVDeposit.depositName = depositName;
+        updatePCVDeposit.underlyingTokenAmount = underlyingTokenAmount;
+        updatePCVDeposit.underlyingToken = underlyingToken;
+
+        emit DepositEdited(index, depositName);
+        emit BalanceUpdate(oldBalance, balance, oldFEIBalance, feiReportBalance);
+    }
+
+    // ----------- Governor only state changing api -----------
+
+    /// @notice function to add a deposit
+    function addDeposit(
+        DepositInfo calldata newPCVDeposit
+    ) external onlyGovernorOrAdmin {
+        _addDeposit(newPCVDeposit);
+    }
+
+    /// @notice function to bulk add deposits
+    function bulkAddDeposits(
+        DepositInfo[] calldata newPCVDeposits
+    ) external onlyGovernorOrAdmin {
+        for (uint256 i = 0; i < newPCVDeposits.length; i++) {
+            _addDeposit(newPCVDeposits[i]);
+        }
+    }
+
+    /// @notice function to edit an existing deposit
+    function editDeposit(
+        uint256 index,
```

Comment:
Really don't like the fact that the only identifier here is index. Would be super easy to accidentally edit the wrong deposit.

Reply:
Disagree. if you're editing the deposits, you are going to query them beforehand, ensure you are modifying the correct index, and then submit your DAO action.